src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +224,428 @@ def _updateImage(self):
+                data_raw_bg_corr = {}
+                # Correct image for background. It must match the polarization (defaulting to MD_POL_NONE).
+                for key in data_raw.keys():
+                    data_raw_bg_corr[key] = self._processBackground(data_raw[key],
+                                                                    data_raw[key].metadata.get(model.MD_POL_MODE,
+                                                                                               model.MD_POL_NONE))
+
+                    if numpy.prod(data_raw_bg_corr[key].shape) > (1280 * 1080):
+                        # TODO not sure this is still true after we changed the triangulation code
+                        # AR conversion fails with very large images due to too much
+                        # memory consumed (> 2Gb). So, rescale + use a "degraded" type that
+                        # uses less memory. As the display size is small (compared
+                        # to the size of the input image, it shouldn't actually
+                        # affect much the output.
+                        logging.info("AR image is very large %s, will convert to "
+                                     "azimuthal projection in reduced precision.",
+                                     data_raw_bg_corr[key].shape)
+                        y, x = data_raw_bg_corr[key].shape
+                        if y > x:
+                            small_shape = 1024, int(round(1024 * x / y))
+                        else:
+                            small_shape = int(round(1024 * y / x)), 1024
+                        # resize
+                        data_raw_bg_corr[key] = img.rescale_hq(data_raw_bg_corr[key], small_shape)
+
+                # convert data into rectangular format (theta-phi-representation)
+                # Check if rectangular converted representation already was calculated for requested ebeam pos.
+                # Note: This calc is very time consuming. Takes about 3.6 sec for one ebeam pos (conversion of 6 images)
+                # and tested on an image of size (256, 1024)
+
+                if ebeam_pos not in polarimetry_cache_raw.keys():
+                    # Get the center wavelength of the filter used (no filter aka "pass-through" use fallback)
+                    # Does not matter from which of the 6 images as they all must have been recorded with the same
+                    # filter -> reuse key
+                    # TODO don't like this solution, key will be the last value of key in the for loop above
+                    # the value of key might be a different one each time...and it is only to get the wl of the filter.
+                    # not sure this is still working in python 3 as I learned today ;)
+                    fallback_wl = 650e-9
+                    band = data_raw_bg_corr[key].metadata.get(model.MD_OUT_WL, fallback_wl)
```

Comment:
You can simplify a little by writing just `....get(model.MD_OUT_WL)` (so it's None if there is no metadata), and remove the `fallback_wl`, so you directly write in the `else`: `wl = 650e-9  # m  (wavelength that cannot be too wrong)`